Entity/Components/HealthComponent.gd

```diff
@@ -41,18 +39,67 @@ func modify_health(amount: float, caster: Entity, is_healing: bool = false) -> v
 	if caster == target: 
 		caster = null
 
-	# apply modification to our health
-	if is_healing:
-		new_health = clampf(current_health + amount, 0, max_health)
-	else:
-		new_health = clampf(current_health - amount, 0, max_health)
-		
+	
+	var leftover_damage: float
+	leftover_damage = amount
+	
+	leftover_damage = block_damage(leftover_damage)
+	_health_damage(leftover_damage)
+
+## removes health without considering block [br]
+## primarily a helper function, but could be used in future for block ignoring attacks[br]
+func _health_damage(damage : int) -> void:
```

Comment:
I would add a check for negative damage here, don't want to heal with negative damage